### PR TITLE
removal of  one paired #ifndef CPL_BYPASS ... #endif in driver-mct/main/cime_comp.mod.F90

### PR DESCRIPTION
**Pull request by fmyuan**
When code building with -DCPL_BYPASS, an option heavily used by ELM modelers, that '#ifndef ... #endif' causes 'mlog' and 'c_mprof_file' not initialized, and later cause unit free error in cime_final(): L3685-3686.

### CODE REVIEW

**Review by peterdschwartz — APPROVED** [review]

**Comment by rljacob** [comment]
Need review from @dmricciuto and @bishtgautam

**Comment by rljacob** [comment]
@dmricciuto please review
@bishtgautam please review.

**Review by bishtgautam — APPROVED** [review]

**Comment by rljacob** [comment]
Waiting on @dmricciuto for review.

**Review by dmricciuto — APPROVED** [review]

**Comment by rljacob** [comment]
@peterdschwartz please start merging this.

**Comment by peterdschwartz** [comment]
pushed to next

**Pull request merged**
